Compile error under Windows/Msys2

I was trying to compile this under a Windows Msys2 environment. It compiles fine, but the linker then exits with an error: `ld.exe: cannot find -lPacket`.

This is probably some boring dependency that's missing, but I *cannot* find which package to install to fix it. `Packet` is an incredibly generic name and very hard to search for. ^^

I should also note that it compiles with `--no-default-features`, meaning that the `bin` feature is causing the error. I tried to narrow it down further, giving me the insight that it is not caused by one of our `bin` dependencies. We only have `env-logger` and `clap`, and removing them does not solve the problem.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,7 +32,7 @@ regex = "1.4.3"
 log = "0.4.13"
 zeroize = { version = "1.2.0", features = ["zeroize_derive"] }
 sodiumoxide = "0.2.6"
-pnet = "0.27.2"
+get_if_addrs = "0.5.3"
 byteorder = "1.4.2"
 anyhow = "1.0.38"
 futures = "0.3.12"

--- a/src/transit.rs
+++ b/src/transit.rs
@@ -23,7 +23,6 @@ use async_std::{
 };
 use futures::{future::TryFutureExt, StreamExt};
 use log::*;
-use pnet::{datalink, ipnetwork::IpNetwork};
 use sodiumoxide::crypto::secretbox;
 use std::{net::ToSocketAddrs, str::FromStr, sync::Arc};
 
@@ -152,11 +151,9 @@ pub async fn init(abilities: Vec<Ability>, relay_url: &RelayUrl) -> Result<Trans
     let mut our_hints: Vec<Hint> = Vec::new();
     if abilities.contains(&Ability::DirectTcpV1) {
         our_hints.extend(
-            datalink::interfaces()
+            get_if_addrs::get_if_addrs()?
                 .iter()
-                .filter(|iface| !datalink::NetworkInterface::is_loopback(iface))
-                .flat_map(|iface| iface.ips.iter())
-                .map(|n| n as &IpNetwork)
+                .filter(|iface| !iface.is_loopback())
                 .map(|ip| {
                     Hint::DirectTcpV1(DirectHint {
                         priority: 0.0,

